sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -87,14 +90,16 @@ internal HttpMessage CreateUploadRequest(string ruleId, string streamName, Reque
         /// <returns></returns>
         internal static IEnumerable<BatchedLogs<T>> Batch<T>(IEnumerable<T> logEntries, UploadLogsOptions options = null)
         {
-            //TODO: use Array pool instead
-            MemoryStream stream = new MemoryStream(SingleUploadThreshold);
-            WriteMemory(stream, BinaryData.FromString("[").ToMemory());
+            // Create an ArrayBufferWriter as backing store for Utf8JsonWriter
+            Azure.Core.ArrayBufferWriter<byte> arrayBuffer = new Azure.Core.ArrayBufferWriter<byte>(SingleUploadThreshold);
+            Utf8JsonWriter writer = new Utf8JsonWriter(arrayBuffer);
+            writer.WriteStartArray();
             int entryCount = 0;
             List<T> currentLogList = new List<T>();
             foreach (var log in logEntries)
             {
                 BinaryData entry;
+                bool isLastEntry = ((entryCount + 1) == logEntries.Count()) ? true : false;
```

Comment:
This will cause us to enumerate to the end of the list each time which could be expensive. Aside from the perf issue, we wouldn't want to cause the IEnumerable to be enumerated multiple times as it could have unintended side-effects depending on how the user has defined their IEnumerable. Instead we should just call ToList at the start of the method and iterate through the list, and use list.Count property.

Reply:
Also, we can just return the value of the equality comparison, i.e.:
```c#
bool isLastEntry = (entryCount + 1 == logEntries.Count);
```